plugins/linguist-backend/src/service/router.ts

```diff
@@ -74,14 +76,16 @@ export async function createRouter(
     await database.getClient(),
   );
 
-  const linguistBackendApi =
+  const catalogClient = new CatalogClient({ discoveryApi: discovery });
+
+  const linguistBackendClient =
     routerOptions.linguistBackendApi ||
-    new LinguistBackendApi(
+    new LinguistBackendClient(
```

Comment:
I think we can avoid creating the `linguistBackendStore` entierly if the user passes in a `linguistBackendApi`.

Could it be perhaps it worth dropping backend from the name? So users can pass in their own `LinguistApi` instead of `LinguistBackendApi`?

Reply:
Sorry, finally got around to this. I'm very confused, I'm not sure what you mean? If I removed `linguistBackendStore` I still need to create it somewhere if I did that in the `linguistBackendApi` I'd still need to bass it the `database` (`DatabaseManager`), right? I feel like I'm missing something obvious.

As fro the name their is already a `LinguistApi` in the frontend so would like to keep this name to make things more clear if that's alright?

Reply:
On line 75 we create the `linguistBackendStore` but we only use it if the user does not pass in their own `linguistBackendApi` so I'm thinking we can avoid creating it unless it is needed. You're right that we still have to pass in the databaseManager but it wouldn't do any work unless needed.